code/modules/power/singularity/emitter.dm

```diff
@@ -18,34 +18,51 @@
 	var/icon_state_on = "emitter_+a"
 	/// The icon state used by the emitter when it's on and low on power.
 	var/icon_state_underpowered = "emitter_+u"
+	///Is the machine active?
 	var/active = FALSE
+	///Does the machine have power?
 	var/powered = FALSE
-	var/fire_delay = 100
-	var/maximum_fire_delay = 100 //10 seconds
-	var/minimum_fire_delay = 20 // 2 seconds
+	///Seconds before the next shot
+	var/fire_delay = 10 SECONDS
+	///Max delay before firing
+	var/maximum_fire_delay = 10 SECONDS
+	///Min delay before firing
+	var/minimum_fire_delay = 2 SECONDS
+	///When was the last shot
 	var/last_shot = 0
+	///Number of shots made (gets reset every few shots)
 	var/shot_number = 0
-	var/welded = FALSE ///if it's welded down to the ground or not. the emitter will not fire while unwelded. if set to true, the emitter will start anchored as well.
+	///if it's welded down to the ground or not. the emitter will not fire while unwelded. if set to true, the emitter will start anchored as well.
+	var/welded = FALSE
+	///Is the emitter id locked?
 	var/locked = FALSE
+	///Used to stop interactions with the object (mainly in the wabbajack statue)
 	var/allow_switch_interact = TRUE
-
+	///What projectile type are we shooting?
 	var/projectile_type = /obj/projectile/beam/emitter/hitscan
+	///What's the projectile sound?
 	var/projectile_sound = 'sound/weapons/emitter.ogg'
+	///Sparks emitted with every shot
 	var/datum/effect_system/spark_spread/sparks
-
+	///Stopes the type of gun we are using inside the emitter
 	var/obj/item/gun/energy/gun
+	///List of all the properties of the inserted gun
 	var/list/gun_properties
-	var/mode = 0
+	//only used to always have the gun properties on non-letal (no other instances found)
+	var/mode = FALSE
 
 	// The following 3 vars are mostly for the prototype
+	///manual shooting?
```

Comment:
what is manual shooting in the context of emitters?

Reply:
uh yeah i should specify, basically you hop onto the emitter and choose the shooting direction, is very janky since you can only shoot at the 8 directions and i don't think is ever used since you can't build those